### ISSUE TITLE
GCP TF Provider Old as dirt

### ISSUE THREAD

**Issue by knee-berts** [fermyon/installer#96]
Howdie! The current google cloud tf provider is set to 3.5 and presents a challenge for users who are using ARM based Macs. Terraform cli took a minute to adopt go 1.16, which supports mac ARM, and that delay cascaded to the GCP provider. Updating the GCP provider to 4.31 will allow for ARM based mac support.

**Comment by knee-berts** [comment]
Happy to take this one. I've got it all running in my own test bed.

**Comment by lann** [comment]
@FrankYang0529 as the original author of the GCP work, any concerns here?

**Comment by FrankYang0529** [comment]
Yeah, I think it's good to upgrade gcp terraform provider. Version `3.5` was released almost 2 years ago. It's good to update to the latest version. Version `4.31` works fine on my laptop.

**Comment by knee-berts** [comment]
Feel free to assign this to me. I have a PR ready to go.
